### PR TITLE
Add filtering and sorting support to GitInspector

### PR DESCRIPTION
**Pull request by radektheloner**

### CODE REVIEW

**Comment by radektheloner** [comment]
@prokopsimek, I haven't added sorting to `getCommits` and filtering nor sorting to `getAuthors` since I did not find such features in GitHub API. But if you think I should add it anyway, I can...

File: src/inspectors/GitInspector.spec.ts [hunk 1]
```diff
@@ -67,6 +67,110 @@ describe('GitInspector', () => {
...
+          sha: expected.hash,
+          date: new Date(expected.date),
+          message: 'msg2\n',
+          author: { name: 'test', email: 'test@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits before a date', async () => {
```
Comment (adelkahomolova) [inline src/inspectors/GitInspector.spec.ts]: Maybe 'until' could be better wording than 'before'

File: src/inspectors/GitInspector.spec.ts [hunk 2]
```diff
@@ -67,6 +67,110 @@ describe('GitInspector', () => {
...
+          sha: expected.hash,
+          date: new Date(expected.date),
+          message: 'msg2\n',
+          author: { name: 'test', email: 'test@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits after a date', async () => {
```
Comment (adelkahomolova) [inline src/inspectors/GitInspector.spec.ts]: Maybe 'since' could be better wording than 'after'

File: src/inspectors/GitInspector.spec.ts (hunk 1 above)
Comment (radektheloner) [reply src/inspectors/GitInspector.spec.ts]: I've borrowed the wording from GitHub docs but doesn't matter... Changed in 
656e4a0.

File: src/inspectors/GitInspector.spec.ts (hunk 2 above)
Comment (radektheloner) [reply src/inspectors/GitInspector.spec.ts]: I've borrowed the wording from GitHub docs but doesn't matter... Changed in
656e4a0.

**Pull request merged**